docs/garbage-collection.md

```diff
@@ -0,0 +1,163 @@
+understands the relationships between these blobs and sets them up using labels
+on the content resources.
+
+Resource labels can also be used to cue the garbage collector on other
+properties, such as expiry, whether an object should be kept without any
+reference or limit what is referenced.
+
+The supported garbage collection labels are...
+
+| Label key | Label value | Supported Resources | Description |
+|---|---|---|---|
+| `containerd.io/gc.root` | _nonempty_ | Content, Snapshots | Keep this object and anything it references. (Clients may set this to a [rfc3339](https://tools.ietf.org/html/rfc3339) timestamp to indicate when this value was set, however, the garbage collector does not parse the value) |
+| `containerd.io/gc.ref.snapshot.<snapshotter>` | `<identifier>` | Content, Snapshots | Resource references the given snapshot `<identifier>` for the snapshotter `<snapshotter>` |
+| `containerd.io/gc.ref.content` | _digest_ | Content, Snapshots, Images, Containers | Resource references the given content blob |
+| `containerd.io/gc.ref.content.<user defined>` | _digest_ | Content, Snapshots, Images, Containers | Resource references the given content blob with a `<user defined>` label key |
+| `containerd.io/gc.expire` | _timestamp_ formatted as [rfc3339](https://tools.ietf.org/html/rfc3339) | Leases | When to expire the lease. The garbage collector will delete the lease after expiration. |
+| `containerd.io/gc.flat` | _nonempty_ | Leases | Ignore label references of leased resources. This only applies when the reference is originating from the lease, if the leased resources are referenced elsewhere, then their label references will be used. |
+
+## Garbage Collection configuration
+
+The garbage collector is scheduled on a background goroutine and runs based on
+a number of configurable factors. By default the garbage collector will attempt
+to keep the database unlocked 98% of the time based on prior calculations of
+lock time from garbage collection. Also by default, the garbage collector will
+not schedule itself if no deletions occurred or after every 100 database writes.
+
+The garbage collection scheduler considers the time the database is locked
+as the pause time. The garbage collection will take longer than this when
+resources are being removed, for example cleaning up snapshots may be slow.
+The scheduler will only schedule after the whole garbage collection is
+completed but use the average pause time for determining when the next run
+attempt is.
+
+Garbage collection may be configured using the `containerd` daemon's
+configuration file, usually at `/etc/containerd/config.toml`. The
+configuration is under the `scheduler` plugin.
+
+### Configuration parameters
+
+| Configuration | Default | Description |
+|---|---|---|
+| `pause_threshold` | 0.02 | Represents the maximum amount of time gc should be scheduled based on the average pause time. A maximum value of .5 (50%) is enforced to prevent over scheduling. |
```

Comment:
this is the first abbreviation for gc.  You really should paren it after the first reference of 'garbage collector'.

Reply:
Done